File: app/Models/CahierTexte.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class CahierTexte extends Model
{
    protected $fillable = [
        'classe_id',
        'professeur_id',
        'matiere_id',
        'date_cours',
        'duree_cours',
        'heure_debut',
        'notion_cours',
        'objectifs',
        'contenu_cours',
        'travail_a_faire',
        'observations'
    ];

    protected $casts = [
        'date_cours' => 'date',
    ];
    
    public function classe()
    {
        return $this->belongsTo(Classe::class);
    }
    
    public function professeur()
    {
        return $this->belongsTo(Professeur::class);
    }

    public function matiere()
    {
        return $this->belongsTo(Matiere::class);
    }

     protected function dateCours(): Attribute
    {
        return Attribute::make(
            get: fn ($value) => \Carbon\Carbon::parse($value),
        );
    }
}

File: professeur_frontend/lib/models/cahier_texte.dart
import 'classe.dart';
import 'professeur.dart';

class CahierTexte {
  final int id;
  final int classeId;
  final int professeurId;
  final DateTime dateCours;
  final int dureeCours;
  final String heureDebut;
  final String notionCours;
  final Classe? classe;
  final String travailAFaire;
  final bool isNonFait;
  final List<int> elevesNonFaitsIds;
  final Professeur? professeur;

  CahierTexte({
    required this.id,
    required this.classeId,
    required this.professeurId,
    required this.dateCours,
    required this.dureeCours,
    required this.heureDebut,
    required this.notionCours,
    this.travailAFaire = '',
    this.isNonFait = false,
    this.elevesNonFaitsIds = const [],
    this.classe,
    this.professeur,
  });

  factory CahierTexte.fromJson(Map<String, dynamic> json) {
    return CahierTexte(
      id: json['id'] ?? 0,
      classeId: json['classe_id'] ?? 0,
      professeurId: json['professeur_id'] ?? 0,
      dateCours: json['date_cours'] != null 
          ? DateTime.parse(json['date_cours']) 
          : DateTime.now(),
      dureeCours: json['duree_cours'] ?? 1,
      heureDebut: json['heure_debut'] ?? '',
      notionCours: json['notion_cours'] ?? '',
      travailAFaire: json['travail_a_faire'] ?? '',
      isNonFait: json['is_non_fait'] ?? false,
      elevesNonFaitsIds: json['eleves_non_faits'] != null
          ? (json['eleves_non_faits'] as List).map<int>((e) => e['id'] as int).toList()
          : [],
      classe: json['classe'] != null ? Classe.fromJson(json['classe']) : null,
      professeur: json['professeur'] != null ? Professeur.fromJson(json['professeur']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'classe_id': classeId,
      'professeur_id': professeurId,
      'date_cours': dateCours.toIso8601String(),
      'duree_cours': dureeCours,
      'heure_debut': heureDebut,
      'notion_cours': notionCours,
      'classe': classe?.toJson(),
      'professeur': professeur?.toJson(),
    };
  }
}